examples/sandbox/src/App.js

```diff
@@ -217,87 +237,129 @@ const config = {
 
 export default function App() {
   return (
-    <SearchProvider config={config}>
-      <WithSearch mapContextToProps={({ wasSearched }) => ({ wasSearched })}>
-        {({ wasSearched }) => {
-          return (
-            <div className="App">
-              <ErrorBoundary>
-                <Layout
-                  header={
-                    <SearchBox
-                      autocompleteMinimumCharacters={3}
-                      autocompleteResults={{
-                        linkTarget: "_blank",
-                        sectionTitle: "Results",
-                        titleField: "title",
-                        urlField: "nps_link",
-                        shouldTrackClickThrough: true,
-                        clickThroughTags: ["test"]
-                      }}
-                      autocompleteSuggestions={true}
-                      debounceLength={0}
-                    />
-                  }
-                  sideContent={
-                    <div>
-                      {wasSearched && (
-                        <Sorting label={"Sort by"} sortOptions={SORT_OPTIONS} />
-                      )}
-                      <Facet
-                        field="states"
-                        label="States"
-                        filterType="any"
-                        isFilterable={true}
-                      />
-                      <Facet
-                        field="world_heritage_site"
-                        label="World Heritage Site"
-                        view={BooleanFacet}
-                      />
-                      <Facet
-                        field="visitors"
-                        label="Visitors"
-                        view={SingleLinksFacet}
-                      />
-                      <Facet
-                        field="date_established"
-                        label="Date Established"
-                        filterType="any"
-                      />
-                      <Facet
-                        field="location"
-                        label="Distance"
-                        filterType="any"
+    <EuiProvider colorMode="light">
+      <SearchProvider config={config}>
+        <WithSearch
+          mapContextToProps={({ wasSearched, authorizeUrl, isLoggedIn }) => ({
```

Comment:
I don't think we should add `isLoggedIn` and `authorizeUrl` to the Search UI state. Since it's specific to the connector, we should probably figure out some other way to pass this information around.

For `authorizeUrl`, you could likely do something like:
```
                        <EuiModalFooter>
                          <EuiButton href={connector.state.authorizeUrl} fill fullWidth>
                            Log in
                          </EuiButton>
                        </EuiModalFooter>
```

You could do something similar for the isLoggedIn state, but let me ponder what that might look like.

Reply:
Maybe something like...

```jsx
const [isLoggedIn, setIsLoggedIn] = useState(connector.state.isLoggedIn);

  useEffect(() => {
    connector.onLoggedInChanged((newIsLoggedIn) => {
      setIsLoggedIn(newIsLoggedIn);
    });
  }, []);
```

Reply:
Yeah, I was agonizing over this one. I chose to go this way to simplify the dev experience: it's much easier to import _everything_ from the Search UI state rather than figuring out what part of data should come from the connector and what part - from the Search UI state. 

But that's not a strong opinion. I can see how we're combining two different things in one object. 

Maybe putting the connector state under a separate field would resolve your concern? Like:

```
mapContextToProps={({ wasSearched, connectorState }) => ({
 
...
...
...
 
!connectorState.isLoggedIn && (
  <EuiModal
    onClose={() => {}}
    maxWidth={354}
    className="login-modal"
  >
```

Let me know if you still prefer to not put it in the Search UI state.

Reply:
Jason's idea is pretty clean as you could provide a hook that customers export

```javascript
  import { useAuthenticationState } from '@elastic/react-search-ui'

   const { isLoggedIn } = useAuthenticationState()
```

I think for now keeping the state within the Search UI is ok but agree there is likely a better route for this. Personally I would vote to keep as is (due to low code impact right now) and then think more in depth how authentication is handled for customers who require authentication in general (I see authentication stretching to not only workplace problems but also app search use-cases where requests require some sort of auth)

Reply:
I'll defer to ya'll on this one. If it eases developer experience then it's not a bad idea I suppose. We could change this API in a future breaking change.

We'll need to document it as new state that will only be populated depending on the connector that you use.

Reply:
Sounds good. I'll add a note to the issue to revisit this.